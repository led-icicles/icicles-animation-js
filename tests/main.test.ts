import Animation from "../icicles/animation";
import Color from "../icicles/color";
import Icicles from "../icicles/icicles";

describe("Animation works correctly", () => {
  test("Encodes name correctly", () => {
    const animation = new Animation("animation", 300);

    expect(animation.getEncodedAnimationName()).toMatchInlineSnapshot(`
      Uint8Array [
        97,
        110,
        105,
        109,
        97,
        116,
        105,
        111,
        110,
        0,
      ]
    `);
  });

  test("VisualFrames works corretly", () => {
    const iciclesSize = 4;
    const icicles = new Icicles(1, iciclesSize);
    const animation = new Animation("animation", icicles.pixels.length, {
      optimize: false,
    });
    for (let i = 0; i < iciclesSize; i++) {
      icicles.setAllPixelsColor(new Color(0, 0, 0));
      icicles.setPixelColorAtIndex(i, new Color(255, 255, 255));
      animation.addFrame(icicles.toFrame(400));
    }

    expect(animation.toFileData()).toMatchSnapshot();
  });

  test("Creates AdditiveFrame corretly", () => {
    const iciclesSize = 4;
    const icicles = new Icicles(1, iciclesSize);
    const animation = new Animation("animation", icicles.pixels.length, {
      optimize: true,
    });
    for (let i = 0; i < iciclesSize; i++) {
      icicles.setAllPixelsColor(new Color(0, 0, 0));
      icicles.setPixelColorAtIndex(i, new Color(255, 255, 255));
      animation.addFrame(icicles.toFrame(400));
    }

    expect(animation.toFileData()).toMatchSnapshot();
  });
});
