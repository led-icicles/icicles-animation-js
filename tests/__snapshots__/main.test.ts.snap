// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animation works correctly Creates AdditiveFrame corretly 1`] = `
Uint8Array [
  97,
  110,
  105,
  109,
  97,
  116,
  105,
  111,
  110,
  0,
  3,
  144,
  1,
  1,
  0,
  0,
  0,
  255,
  255,
  255,
  2,
  144,
  1,
  0,
  0,
  0,
  255,
  255,
  255,
  0,
  0,
  0,
  0,
  0,
  0,
  2,
  144,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  255,
  255,
  255,
  0,
  0,
  0,
  2,
  144,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  255,
  255,
  255,
]
`;

exports[`Animation works correctly VisualFrames works corretly 1`] = `
Uint8Array [
  97,
  110,
  105,
  109,
  97,
  116,
  105,
  111,
  110,
  0,
  2,
  144,
  1,
  255,
  255,
  255,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  2,
  144,
  1,
  0,
  0,
  0,
  255,
  255,
  255,
  0,
  0,
  0,
  0,
  0,
  0,
  2,
  144,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  255,
  255,
  255,
  0,
  0,
  0,
  2,
  144,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  255,
  255,
  255,
]
`;
